src/rpc/server.ts

```diff
@@ -977,10 +977,17 @@ export class RpcServer {
         `${friendbotUrl}?addr=${encodeURIComponent(account)}`
       );
 
-      const meta = xdr.TransactionMeta.fromXDR(
-        response.data.result_meta_xdr,
-        'base64'
-      );
+      let meta: xdr.TransactionMeta;
+      if (!response.data.result_meta_xdr) {
```

Comment:
is it worth retaining the conditional check for meta from friendbot for performance aspect? Since response data can vary, could be time to just refactor/reduce code to use rpc call since it is deterministic in all cases.

Reply:
Yeahhh I dunno, I was conflicted on this; initially I just used RPC. I don't have a strong opinion: it's nice to have a faster return but also the code gets harder to read for minimal benefit.

Reply:
Imma merge it this way because backporting the tests to not support the friendbot parsing method will be a massive pain in the ass lol